test/unit/agent/synthetics.test.js

```diff
@@ -5,41 +5,45 @@
 
 'use strict'
 
-const tap = require('tap')
+const test = require('node:test')
+const assert = require('node:assert')
+const promiseResolvers = require('../../lib/promise-resolvers')
 const helper = require('../../lib/agent_helper')
 
-tap.test('synthetics transaction traces', (t) => {
-  t.autoend()
-
-  let agent
-
-  t.beforeEach(() => {
-    agent = helper.loadMockedAgent({
+test('synthetics transaction traces', async (t) => {
+  t.beforeEach((ctx) => {
+    ctx.nr = {}
+    ctx.nr.agent = helper.loadMockedAgent({
       trusted_account_ids: [357]
     })
   })
 
-  t.afterEach(() => {
-    helper.unloadAgent(agent)
+  t.afterEach((ctx) => {
+    helper.unloadAgent(ctx.nr.agent)
   })
 
-  t.test('should include synthetic intrinsics if header is set', (t) => {
-    helper.runInTransaction(agent, function (txn) {
-      txn.syntheticsData = {
+  await t.test('should include synthetic intrinsics if header is set', async (t) => {
+    const { agent } = t.nr
+    const { promise, resolve } = promiseResolvers()
```

Comment:
Agreed. It's easier in Node >=22 because it's native, but until then, you can try to make this go somewhere https://x.com/jsumners79/status/1813939927169827248